src/freenet/clients/fcp/ClientGet.java

```diff
@@ -183,7 +183,7 @@ public ClientGet(PersistentRequestClient globalClient, FreenetURI uri, boolean d
 			        targetFile.delete();
                     Logger.error(this, "Target file already exists but is zero length, deleting...");
 			    }
-			    if(targetFile.exists())
+			    else
 			        throw new IOException("Target filename exists already: "+targetFile);
```

Comment:
I'm sorry, this is not a valid change. In the existing code, an `IOException` is thrown when the file existed, was empty, but could not be deleted. This commit changes that behaviour, leaving a (probably unwritable) existing file in place without exception.